torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -3117,6 +3118,78 @@ Tensor triangular_solve_jvp(
   );
 }
 
+Tensor linalg_solve_triangular_forward_AD(
+    const Tensor& A_t,
+    const Tensor& B_t,
+    const Tensor& A,
+    const Tensor& X,
+    const bool upper,
+    const bool left,
+    const bool unitriangular) {
+  // The forward AD formula (for left = true) is A^{-1}(B_t - A_tX)
+  // For the derivation see:
+  // [Note: Forward / Backward AD solve_triangular]
+  const Tensor proj_A_t = upper ? A_t.triu(static_cast<int>(unitriangular))
+                                : A_t.tril(- static_cast<int>(unitriangular));
+  const Tensor X_t = B_t - (left ? at::matmul(proj_A_t, X) : at::matmul(X, proj_A_t));
+  return at::native::linalg_solve_triangular(A, X_t, upper, left, unitriangular);
```

Comment:
This call breaks some mobile builds internally on android. Sorry for the lack of external CI signal on this still working on improving those. Can you change to 'at::linalg_solve_triangular(A, X_t, upper, left, unitriangular);' instead of calling through native directly?

Reply:
Oh, I did not know that was problematic in forward AD, as it's not going to be differentiated.
I will make sure we take this into account when implementing similar AD formulas. cc @IvanYashchuk @nikitaved @albanD

Reply:
@JacobSzwejbka do you have some more details on why this is not feasible?
Calling directly into native is a (small) perf optimization we do quite a bit.